libr/anal/passmanager/passmanager.inc.c

```diff
@@ -0,0 +1,185 @@
+typedef struct Pr_(TYPE_NAME) {
+        PASS_(TYPE_NAME) *p;
+        HtPP *passResults; /*<void *object, void *result>*/
+        PM_(TYPE_NAME) *parent;
+} PR_(TYPE_NAME);
+
+/*
+ * passes: <char *Pass->name, PassRunner>
+ * running: <TYPE *object,HtPP<char *passName, NULL>>. Mainly used
+ * to detect circular dependencies
+ */
+typedef struct Pm_(TYPE_NAME) {
+	HtPP *passes;
+	HtPP *running;
+	RAnal *parent;
+} PM_(TYPE_NAME);
+
+static PR_(TYPE_NAME) *PMFN_(passrunner_new)(PASS_(TYPE_NAME) *p) {
+	PR_(TYPE_NAME) *pr = R_NEW0 (PR_(TYPE_NAME));
+	r_return_val_if_fail(pr, NULL);
```

Comment:
we dont check heap pointers with an asswer, but we do that with function arguments (*p), you should use a normal if() here. the reasoning is:

* the heap can fail at anytime, but we dont want r2 to die if the system is under pressure and cant allocate memory
* if a function of these gets a NULL as argument, thats something it shouldnt happen, so we can check that in developer builds with an assert message, but release builds will directly crash if that happens (but it shouldnt because we tested it in dev builds)